### DISCUSSION TITLE
Feature Request - Option for alternative G code format: G2 Xn Yn In Jn

### DISCUSSION THREAD

**Discussion by printhefuture** [FormerLurker/ArcWelderPlugin#168] [Ideas]
Is it possible to add code, and a user selectable option, for the G2 / G3 code outputs to have the[ alternative format](https://marlinfw.org/docs/gcode/G002-G003.html), which uses _X, I, J_ instead of _X, I, R_?

E.g. 
**Current Output:**       G3 X2 Y7 R5
**Alternative Output:**  G3 X2 Y7 I-4 J-3

The benefit will be saving many processor cycles on boards without FPUs. This is due to the current code in Marlin's G2_G3.cpp (v2.07). Every time a G2 / G3 is encountered, it does alot of expensive math ops, these ops aren't needed with the alternative format.  

The (small) downside of this request - apart from you coding it - is a slight increase in processing time when ArcWeld does it's business. 

As an aside, I'm currently re-writing G2_G3.cpp with a bunch of other optimisations for boards that don't have FPUs.

Thanks!

**Comment by FormerLurker** [comment]
Are you sure you don't have it backwards?  ArcWelder always outputs X Y I J now.  It would be extremely easy to output the X Y R form, especially since I already have to calculate the radius.  In fact, I actually convert R to I J as a final step.

Now, the X Y R form has less support, but I don't see any problem adding an option for it.  It will reduce the gcode size even further if it's supported.
